test/options.test.ts

```diff
@@ -37,8 +37,11 @@ import {
   Options,
   defaultPropagatorFactory,
 } from '../src/options';
+import * as utils from './utils';
 
 describe('options', () => {
+  beforeEach(utils.cleanEnvironment);
```

Comment:
let's clean after tests, not before, this suggests that we don't know where the "messy" environment settings come from

Reply:
> suggests that we don't know where the "messy" environment settings come from
Well... we don't. And I feel bad for it honestly and would focus on making sure our tests run in a clean environment.

The environment is controlled by the runner. We don't assure the order of the tests executed so any of our tests might be the first, if your personal environment contains `OTEL_` prefixed stuff it already is affecting our tests - we should make sure we avoid falling into that.

Then... taking into account that we have to do before either way I don't see a reason to do it after as well.

Reply:
Yeah if you think that's the right way to go then merge.